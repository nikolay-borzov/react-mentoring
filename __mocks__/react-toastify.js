import * as toastify from 'react-toastify'

toastify.toast.error = jest.fn()

export * from 'react-toastify'
