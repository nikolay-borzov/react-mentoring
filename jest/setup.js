import promiseFinally from 'promise.prototype.finally'

import Enzyme, { shallow, render, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

// Bring '.finally' to Promise (only for Node 8). Will be available in Node 10
promiseFinally.shim()

Enzyme.configure({ adapter: new Adapter() })

// Make Enzyme functions available in all test files without importing
global.shallow = shallow
global.render = render
global.mount = mount
