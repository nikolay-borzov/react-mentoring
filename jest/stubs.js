export const film = {
  id: 1,
  title: 'Film title',
  poster_path: 'http://via.placeholder.com/500x750',
  overview: 'Film overview',
  release_date: '2018-05-04',
  genres: ['some genre', 'another genre']
}

export const films = [film]
