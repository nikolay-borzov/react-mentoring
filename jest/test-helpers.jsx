import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { runSaga as runSagaOriginal } from 'redux-saga'
import configureMockStore from 'redux-mock-store'

/**
 * Creates test closure for snapshot testing
 * @deprecated Use itRendersCorrectlyShallow
 * @param {Function} getComponent Function returning component to test snapshot
 * @param {string} [testName='renders correctly']
 */
export const itRendersCorrectly = (
  getComponent,
  testName = 'renders correctly'
) => {
  it(testName, () => {
    const tree = renderer.create(getComponent()).toJSON()

    expect(tree).toMatchSnapshot()
  })
}

/**
 * Creates test closure for snapshot testing (using Enzyme shallow)
 * @param {Function} getComponent Function returning component to test snapshot
 * @param {string} [testName='renders correctly']
 */
export const itRendersCorrectlyShallow = (
  getComponent,
  testName = 'renders correctly'
) => {
  it(testName, () => {
    const wrapper = shallow(getComponent())

    expect(wrapper).toMatchSnapshot()
  })
}

/**
 * Creates test closure to check whether component contains another component
 * @param {Function} getComponent Function returning parent component
 * @param {string} componentDisplayName Child component display name
 * @param {Object} params Optional params
 * @param {number} [params.testName=`contains ${componentDisplayName} component`]
 * @param {number} [params.expectedCount=1] Expected number of children components
 * @param {Object} params.expectedProps First child component properties map
 */
export const itContainsComponent = (
  getComponent,
  componentDisplayName,
  { testName, expectedCount = 1, expectedProps } = {}
) => {
  it(testName || `contains ${componentDisplayName} component`, () => {
    const wrapper = shallow(getComponent())
    const children = wrapper.find(componentDisplayName)

    expect(children.length).toBe(expectedCount)
    if (expectedProps) {
      expect(children.first().props()).toEqual(expectedProps)
    }
  })
}

/**
 * Creates mocked store
 */
export const getMockStore = () => configureMockStore([])

/**
 * Sets relative URL
 * @param {string} url
 */
export const setUrl = url => {
  window.history.pushState({}, '', url)
}

/**
 * Runs saga and collects dispatched actions
 */
export const runSaga = (saga, state = {}, arg1 = undefined) => {
  const dispatched = []
  return runSagaOriginal(
    {
      dispatch: action => dispatched.push(action),
      getState: () => state
    },
    saga,
    arg1
  )
    .toPromise()
    .then(() => dispatched)
}
