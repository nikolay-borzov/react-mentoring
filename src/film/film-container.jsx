import React from 'react'

import { Header } from '../core/header'
import { Footer } from '../core/footer'
import { SiteName } from '../core/site-name'

export class FilmContainer extends React.Component {
  render() {
    return (
      <React.Fragment>
        <Header>
          <SiteName />
        </Header>
        <Footer />
      </React.Fragment>
    )
  }
}
