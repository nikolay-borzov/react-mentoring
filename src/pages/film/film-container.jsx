import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { toast } from 'react-toastify'

import {
  fetchFilm,
  setRelatedFilmsSearchParams,
  fetchRelatedFilms,
  selectors
} from '../../redux/modules/view'

import {
  Header,
  Footer,
  SiteName,
  LoadingBlock,
  ToastError,
  FilmsGrid,
  SearchResultsPanel,
  ContentMessage
} from '../../components'

import { FilmDetails } from './components/film-details'

const mapStateToProps = state => ({
  film: selectors.film.film(state),
  filmIsFetching: selectors.film.isFetching(state),
  genre: selectors.relatedFilms.searchParams.search(state),
  relatedFilms: selectors.relatedFilms.films.films(state),
  relatedFilmsIsFetching: selectors.relatedFilms.films.isFetching(state),
  relatedFilmsError: selectors.relatedFilms.films.error(state)
})

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      fetchFilm,
      setRelatedFilmsSearchParams,
      fetchRelatedFilms
    },
    dispatch
  )

export class FilmContainer extends React.Component {
  static propTypes = {
    filmId: PropTypes.number.isRequired,
    film: PropTypes.object,
    filmIsFetching: PropTypes.bool.isRequired,
    genre: PropTypes.string.isRequired,
    relatedFilms: PropTypes.arrayOf(PropTypes.object),
    relatedFilmsIsFetching: PropTypes.bool.isRequired,
    relatedFilmsError: PropTypes.object,
    fetchFilm: PropTypes.func.isRequired,
    setRelatedFilmsSearchParams: PropTypes.func.isRequired,
    fetchRelatedFilms: PropTypes.func.isRequired
  }

  componentDidMount() {
    this.loadFilm(this.props.filmId)
  }

  loadFilm(id) {
    return this.props
      .fetchFilm(id)
      .then(() => {
        const genres = this.props.film.genres
        this.loadRelatedFilms(genres[0])
      })
      .catch(error => {
        toast.error(
          <ToastError message="Unable to load the movie" error={error} />
        )
      })
  }

  loadRelatedFilms(genre) {
    const { setRelatedFilmsSearchParams, fetchRelatedFilms } = this.props

    setRelatedFilmsSearchParams({
      search: genre
    })

    return fetchRelatedFilms().catch(error => {
      this.setState({
        relatedFilmsError: true
      })

      toast.error(
        <ToastError message="Unable to load related movies" error={error} />
      )
    })
  }

  render() {
    const {
      film,
      filmIsFetching,
      genre,
      relatedFilms,
      relatedFilmsIsFetching,
      relatedFilmsError
    } = this.props

    let relatedFilmsBlock

    if (relatedFilms.length > 0) {
      relatedFilmsBlock = <FilmsGrid films={relatedFilms} />
    } else if (relatedFilmsError) {
      relatedFilmsBlock = (
        <ContentMessage className="error-message">
          Unable to load related movies
        </ContentMessage>
      )
    } else {
      relatedFilmsBlock = (
        <ContentMessage className="font-size-header font-bold color-alt">
          No related movies found
        </ContentMessage>
      )
    }

    return (
      <React.Fragment>
        <Header>
          <div className="padding-controls flex flex-align-center">
            <div className="flex-grow">
              <SiteName />
            </div>
            <a href="/" className="button button--small button--primary">
              Search
            </a>
          </div>
          <LoadingBlock isLoaded={!filmIsFetching} hideText={true}>
            <FilmDetails film={film} />
          </LoadingBlock>
        </Header>

        <main className="content">
          <LoadingBlock isLoaded={!relatedFilmsIsFetching}>
            <SearchResultsPanel>Films by {genre} genre</SearchResultsPanel>
            {relatedFilmsBlock}
          </LoadingBlock>
        </main>
        <Footer />
      </React.Fragment>
    )
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(FilmContainer)
