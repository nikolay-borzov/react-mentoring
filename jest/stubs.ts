import { Film } from '../src/entities/film'

export const film: Film = {
  id: 1,
  title: 'Film title',
  tagline: 'Put that cookie down',
  vote_average: 7.5,
  vote_count: 667,
  release_date: '2018-05-04',
  poster_path: 'https://picsum.photos/g/300/450/?random',
  overview: 'Once upon a time...',
  budget: 15000000,
  revenue: 59735548,
  genres: ['some genre', 'another genre']
}

export const films = [film]

function getRandomImageId() {
  // https://picsum.photos/images
  return Math.floor(Math.random() * Math.floor(1084))
}

export function getRandomFilm(id: number) {
  return {
    ...film,
    id,
    poster_path: film.poster_path.replace(
      'random',
      `image=${getRandomImageId()}`
    )
  }
}
