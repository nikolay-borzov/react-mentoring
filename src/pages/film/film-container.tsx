import React from 'react'
import { connect } from 'react-redux'
import { Helmet } from 'react-helmet'
import { toast } from 'react-toastify'

import { fetchFilm, reFetchFilms, selectors } from '../../redux/modules/view'
import { GetLoadable } from '../../components/helpers'
import {
  Header,
  Footer,
  SiteName,
  LoadingBlock,
  ToastError,
  SearchResultsPanel,
  FilmsGridProps,
  ContentMessageProps
} from '../../components'
import { NavigateButton } from '../../styles'
import { FilmDetails } from './components/film-details'
import { Film } from '../../entities/film'
import { AppState } from '../../redux/reducer'

/* istanbul ignore next */
const FilmsGrid = GetLoadable<FilmsGridProps>(() =>
  import('../../components/films-grid/films-grid')
)
/* istanbul ignore next */
const ContentMessage = GetLoadable<ContentMessageProps>(() =>
  import('../../components/content-message')
)
/* istanbul ignore next */
const NotFound = GetLoadable(() => import('../not-found'))

const mapStateToProps = (state: AppState) => ({
  film: selectors.film.film(state),
  filmIsFetching: selectors.film.isFetching(state),
  filmError: selectors.film.error(state),
  genre: selectors.relatedFilms.searchParams.search(state),
  relatedFilms: selectors.relatedFilms.films.films(state),
  relatedFilmsIsFetching: selectors.relatedFilms.films.isFetching(state),
  relatedFilmsError: selectors.relatedFilms.films.error(state)
})

const mapDispatchToProps = {
  fetchFilm,
  reFetchFilms
}

export interface FilmContainerProps {
  match: {
    params: { id: string }
  }
  film?: Film
  filmIsFetching: boolean
  filmError?: Error
  genre?: string
  relatedFilms: Film[]
  relatedFilmsIsFetching: boolean
  relatedFilmsError?: Error
  fetchFilm(id: number): void
  reFetchFilms(): void
}

export class FilmContainer extends React.Component<FilmContainerProps> {
  constructor(props: FilmContainerProps) {
    super(props)

    if (IS_SERVER) {
      this.initialLoad(props)
    }
  }

  initialLoad(props: FilmContainerProps) {
    this.loadFilm(props.match.params.id)
  }

  componentDidMount() {
    const film = this.props.film
    const filmId = parseInt(this.props.match.params.id, 10)
    // If films was loaded on server - load related films
    if (film && film.id === filmId) {
      this.props.reFetchFilms()
    } else {
      this.initialLoad(this.props)
    }
  }

  componentDidUpdate(prevProps: FilmContainerProps) {
    const filmId = this.props.match.params.id
    const { filmError, relatedFilmsError } = this.props
    // Load film if film id has changed
    if (prevProps.match.params.id !== filmId) {
      this.loadFilm(filmId)
    }

    if (filmError) {
      toast.error(
        <ToastError message="Unable to load the movie" error={filmError} />
      )
    } else if (relatedFilmsError) {
      toast.error(
        <ToastError
          message="Unable to load related movies"
          error={relatedFilmsError}
        />
      )
    }
  }

  loadFilm(id: string) {
    this.props.fetchFilm(parseInt(this.props.match.params.id, 10))
  }

  render() {
    const {
      film,
      filmIsFetching,
      genre,
      relatedFilms,
      relatedFilmsIsFetching,
      relatedFilmsError
    } = this.props

    if (film && !film.id) {
      return <NotFound />
    }

    let relatedFilmsBlock = null

    if (relatedFilms.length > 0) {
      relatedFilmsBlock = <FilmsGrid films={relatedFilms} />
    } else if (relatedFilmsError) {
      relatedFilmsBlock = (
        <ContentMessage className="error-message">
          Unable to load related movies
        </ContentMessage>
      )
    } else if (film) {
      relatedFilmsBlock = (
        <ContentMessage className="font-size-header font-bold color-alt">
          No related movies found
        </ContentMessage>
      )
    }

    return (
      <>
        {film ? (
          <Helmet>
            <title>{film.title} ðŸŽ¥ Movie Search</title>
          </Helmet>
        ) : null}
        <Header>
          <div className="padding-controls flex flex-align-center">
            <div className="flex-grow">
              <SiteName />
            </div>
            <NavigateButton to="/search" primary small>
              Search
            </NavigateButton>
          </div>
          <LoadingBlock isLoaded={!filmIsFetching} hideText={true}>
            <FilmDetails film={film} />
          </LoadingBlock>
        </Header>

        <main className="content">
          <LoadingBlock isLoaded={!relatedFilmsIsFetching}>
            <SearchResultsPanel>Films by {genre} genre</SearchResultsPanel>
            {relatedFilmsBlock}
          </LoadingBlock>
        </main>
        <Footer />
      </>
    )
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(FilmContainer)
