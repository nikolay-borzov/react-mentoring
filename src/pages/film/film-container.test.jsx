import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import { film, films } from '../../../jest/stubs'

import { FilmContainer } from './film-container'

import filmService from '../../services/film-service'
import { toast } from 'react-toastify'

jest.mock('../../services/film-service')

jest.mock('react-toastify', () => ({
  toast: {
    error: jest.fn()
  }
}))

// TODO: Mock ContentImage component https://medium.com/xebia/de-mystifying-jest-snapshot-test-mocks-8e7183d109ea

describe('FilmContainer page component', () => {
  const getFilmsResponse = {
    data: films
  }

  let props = {
    filmId: film.id
  }

  const render = () => {
    const wrapper = shallow(<FilmContainer {...props} />)

    return { wrapper, instance: wrapper.instance() }
  }

  beforeEach(() => {
    filmService.getFilm.mockClear()
    filmService.getFilms.mockClear()
  })

  describe('it renders correctly', () => {
    let tree

    beforeEach(() => {
      filmService.getFilm.mockReturnValue(Promise.resolve(film))
      filmService.getFilms.mockReturnValue(Promise.resolve(getFilmsResponse))
      tree = renderer.create(<FilmContainer {...props} />)
    })

    it('when film and relatedFilms are provided', () => {
      // TODO: Perhaps there is a better way to get data rendered
      // https://github.com/facebook/jest/issues/2157
      // Set data directly to avoid waiting for promise resolving
      tree.root.instance.setState({
        film,
        isFilmLoaded: true,
        relatedFilms: films,
        isRelatedFilmsLoaded: true
      })

      // TODO: Mock ContentImage
      expect(tree.toJSON()).toMatchSnapshot()
    })

    it('when the film is unavailable', () => {
      tree.root.instance.setState({
        film: null,
        isFilmLoaded: true,
        relatedFilms: films,
        isRelatedFilmsLoaded: true
      })

      expect(tree.toJSON()).toMatchSnapshot()
    })
  })

  it('loads a film by id and related films by genre', () => {
    filmService.getFilm.mockReturnValue(Promise.resolve(film))

    const { wrapper, instance } = render()

    expect(filmService.getFilm).toHaveBeenCalled()

    return instance.loadFilm(film.id).then(() => {
      const state = wrapper.state()

      expect(state.film).toBe(film)
      expect(state.genre).toBe(film.genres[0])
      expect(state.isFilmLoaded).toBe(true)

      expect(filmService.getFilms).toHaveBeenCalled()
    })
  })

  it('displays error when unable to load the film', async () => {
    const error = new Error('Film load error')
    filmService.getFilm.mockReturnValue(Promise.reject(error))

    const { wrapper, instance } = render()

    expect.assertions(2)

    return instance.loadFilm(film.id).then(() => {
      expect(toast.error).toHaveBeenCalled()
      expect(wrapper.state().isFilmLoaded).toBe(true)
    })
  })

  it('loads related films by genre', () => {
    filmService.getFilm.mockReturnValue(Promise.resolve(film))
    filmService.getFilms.mockReturnValue(Promise.resolve(getFilmsResponse))

    const { wrapper, instance } = render()

    return instance.loadRelatedFilms(film.genres[0]).then(() => {
      const state = wrapper.state()

      expect(filmService.getFilms).toHaveBeenCalledWith(instance.queryParams)

      expect(state.relatedFilms).toBe(films)
      expect(state.isRelatedFilmsLoaded).toBe(true)
    })
  })

  it('displays error when unable to load related films', async () => {
    const error = new Error('Films load error')
    filmService.getFilms.mockReturnValue(Promise.reject(error))

    const { wrapper, instance } = render()

    expect.assertions(2)

    return instance.loadRelatedFilms(film.genres[0]).then(() => {
      expect(toast.error).toHaveBeenCalled()
      expect(wrapper.state().isRelatedFilmsLoaded).toBe(true)
    })
  })
})
