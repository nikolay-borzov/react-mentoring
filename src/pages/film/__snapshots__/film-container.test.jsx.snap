// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmContainer page component renders correctly when connected to the store 1`] = `
<FilmContainer
  fetchFilm={[Function]}
  film={null}
  filmError={null}
  filmIsFetching={false}
  genre=""
  reFetchFilms={[Function]}
  relatedFilms={Array []}
  relatedFilmsError={null}
  relatedFilmsIsFetching={false}
  store={
    Object {
      "close": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "runSaga": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "close": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "runSaga": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;

exports[`FilmContainer page component renders correctly when film and relatedFilms are provided 1`] = `
<React.Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Film title
       ðŸŽ¥ Movie Search
    </title>
  </HelmetWrapper>
  <Header>
    <div
      className="padding-controls flex flex-align-center"
    >
      <div
        className="flex-grow"
      >
        <SiteName />
      </div>
      <Button
        primary={true}
        small={true}
        to="/search"
      >
        Search
      </Button>
    </div>
    <LoadingBlock
      hideText={true}
      isLoaded={true}
    >
      <FilmDetails
        film={
          Object {
            "genres": Array [
              "some genre",
              "another genre",
            ],
            "id": 1,
            "overview": "Film overview",
            "poster_path": "https://picsum.photos/g/300/450/?random",
            "release_date": "2018-05-04",
            "title": "Film title",
            "vote_average": 7.5,
          }
        }
      />
    </LoadingBlock>
  </Header>
  <main
    className="content"
  >
    <LoadingBlock
      hideText={false}
      isLoaded={true}
    >
      <SearchResultsPanel>
        Films by 
        Horror
         genre
      </SearchResultsPanel>
      <LoadableComponent
        films={
          Array [
            Object {
              "genres": Array [
                "some genre",
                "another genre",
              ],
              "id": 1,
              "overview": "Film overview",
              "poster_path": "https://picsum.photos/g/300/450/?random",
              "release_date": "2018-05-04",
              "title": "Film title",
              "vote_average": 7.5,
            },
          ]
        }
      />
    </LoadingBlock>
  </main>
  <Footer />
</React.Fragment>
`;

exports[`FilmContainer page component renders correctly when no film found nor related films found  1`] = `
<React.Fragment>
  <Header>
    <div
      className="padding-controls flex flex-align-center"
    >
      <div
        className="flex-grow"
      >
        <SiteName />
      </div>
      <Button
        primary={true}
        small={true}
        to="/search"
      >
        Search
      </Button>
    </div>
    <LoadingBlock
      hideText={true}
      isLoaded={true}
    >
      <FilmDetails
        film={null}
      />
    </LoadingBlock>
  </Header>
  <main
    className="content"
  >
    <LoadingBlock
      hideText={false}
      isLoaded={true}
    >
      <SearchResultsPanel>
        Films by 
        Horror
         genre
      </SearchResultsPanel>
    </LoadingBlock>
  </main>
  <Footer />
</React.Fragment>
`;

exports[`FilmContainer page component renders correctly when no related films found 1`] = `
<React.Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Film title
       ðŸŽ¥ Movie Search
    </title>
  </HelmetWrapper>
  <Header>
    <div
      className="padding-controls flex flex-align-center"
    >
      <div
        className="flex-grow"
      >
        <SiteName />
      </div>
      <Button
        primary={true}
        small={true}
        to="/search"
      >
        Search
      </Button>
    </div>
    <LoadingBlock
      hideText={true}
      isLoaded={true}
    >
      <FilmDetails
        film={
          Object {
            "genres": Array [
              "some genre",
              "another genre",
            ],
            "id": 1,
            "overview": "Film overview",
            "poster_path": "https://picsum.photos/g/300/450/?random",
            "release_date": "2018-05-04",
            "title": "Film title",
            "vote_average": 7.5,
          }
        }
      />
    </LoadingBlock>
  </Header>
  <main
    className="content"
  >
    <LoadingBlock
      hideText={false}
      isLoaded={true}
    >
      <SearchResultsPanel>
        Films by 
        Horror
         genre
      </SearchResultsPanel>
      <LoadableComponent
        className="font-size-header font-bold color-alt"
      >
        No related movies found
      </LoadableComponent>
    </LoadingBlock>
  </main>
  <Footer />
</React.Fragment>
`;

exports[`FilmContainer page component renders correctly when related films had failed to load 1`] = `
<React.Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Film title
       ðŸŽ¥ Movie Search
    </title>
  </HelmetWrapper>
  <Header>
    <div
      className="padding-controls flex flex-align-center"
    >
      <div
        className="flex-grow"
      >
        <SiteName />
      </div>
      <Button
        primary={true}
        small={true}
        to="/search"
      >
        Search
      </Button>
    </div>
    <LoadingBlock
      hideText={true}
      isLoaded={true}
    >
      <FilmDetails
        film={
          Object {
            "genres": Array [
              "some genre",
              "another genre",
            ],
            "id": 1,
            "overview": "Film overview",
            "poster_path": "https://picsum.photos/g/300/450/?random",
            "release_date": "2018-05-04",
            "title": "Film title",
            "vote_average": 7.5,
          }
        }
      />
    </LoadingBlock>
  </Header>
  <main
    className="content"
  >
    <LoadingBlock
      hideText={false}
      isLoaded={true}
    >
      <SearchResultsPanel>
        Films by 
        Horror
         genre
      </SearchResultsPanel>
      <LoadableComponent
        className="error-message"
      >
        Unable to load related movies
      </LoadableComponent>
    </LoadingBlock>
  </main>
  <Footer />
</React.Fragment>
`;

exports[`FilmContainer page component renders correctly when the film is not found 1`] = `<LoadableComponent />`;

exports[`FilmContainer page component renders correctly when the film is unavailable 1`] = `
<React.Fragment>
  <Header>
    <div
      className="padding-controls flex flex-align-center"
    >
      <div
        className="flex-grow"
      >
        <SiteName />
      </div>
      <Button
        primary={true}
        small={true}
        to="/search"
      >
        Search
      </Button>
    </div>
    <LoadingBlock
      hideText={true}
      isLoaded={true}
    >
      <FilmDetails
        film={null}
      />
    </LoadingBlock>
  </Header>
  <main
    className="content"
  >
    <LoadingBlock
      hideText={false}
      isLoaded={true}
    >
      <SearchResultsPanel>
        Films by 
        Horror
         genre
      </SearchResultsPanel>
      <LoadableComponent
        films={
          Array [
            Object {
              "genres": Array [
                "some genre",
                "another genre",
              ],
              "id": 1,
              "overview": "Film overview",
              "poster_path": "https://picsum.photos/g/300/450/?random",
              "release_date": "2018-05-04",
              "title": "Film title",
              "vote_average": 7.5,
            },
          ]
        }
      />
    </LoadingBlock>
  </main>
  <Footer />
</React.Fragment>
`;
